atc/engine/check_delegate.go

```diff
@@ -111,25 +118,56 @@ func (d *checkDelegate) WaitToRun(ctx context.Context, scope db.ResourceConfigSc
 		}
 	}
 
-	end, err := scope.LastCheckEndTime()
-	if err != nil {
-		if releaseErr := lock.Release(); releaseErr != nil {
-			logger.Error("failed-to-release-lock", releaseErr)
+	shouldRun := false
+	if !d.plan.IsPeriodic() {
```

Comment:
i know i suggested this name, but i just noticed it might be a little confusing - `IsPeriodic` and `IsManuallyTriggered` sound mutually exclusive, but they're not. i can't really think of a better name though. if you have any ideas by all means - won't block the PR on this either way, since someone can always rename it later.

Reply:
Yeah, I feel the same. Sorry for blindly accepting your renaming suggestion. I just renamed the method to `IsStepEmbedded()`.

Reply:
it's possible to be `IsPeriodic && IsManuallyTriggered` which is the case this PR addresses (a webhook-triggered check for a 'periodic' check plan). so for someone that doesn't know what those actually mean it might seem a little confusing. no big deal though.